NullPointerException if a system variable is present in homegear

As soon as I add a system variable in homegear I see the following exception in the openhab log when scanning for new things in the homematic binding in paper ui:
```
2016-11-19 03:08:43.152 [ERROR] [very.HomematicDeviceDiscoveryService] - 
java.lang.NullPointerException
        at org.openhab.binding.homematic.type.MetadataUtils.createBigDecimal(MetadataUtils.java:251)[189:org.openhab.binding.homematic:2.0.0.b4]
        at org.openhab.binding.homematic.type.HomematicTypeGeneratorImpl.createChannelType(HomematicTypeGeneratorImpl.java:242)[189:org.openhab.binding.homematic:2.0.0.b4]
        at org.openhab.binding.homematic.type.HomematicTypeGeneratorImpl.generate(HomematicTypeGeneratorImpl.java:128)[189:org.openhab.binding.homematic:2.0.0.b4]
        at org.openhab.binding.homematic.handler.HomematicBridgeHandler.onDeviceLoaded(HomematicBridgeHandler.java:277)[189:org.openhab.binding.homematic:2.0.0.b4]
        at org.openhab.binding.homematic.internal.communicator.AbstractHomematicGateway.loadAllDeviceMetadata(AbstractHomematicGateway.java:389)[189:org.openhab.binding.homematic:2.0.0.b4]
        at org.openhab.binding.homematic.discovery.HomematicDeviceDiscoveryService$1.run(HomematicDeviceDiscoveryService.java:108)[189:org.openhab.binding.homematic:2.0.0.b4]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_65]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_65]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)[:1.8.0_65]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)[:1.8.0_65]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_65]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_65]
        at java.lang.Thread.run(Thread.java:745)[:1.8.0_65]
```

add a variable:
```   
homegear -e 'runcommand $hg->setSystemVariable("maxValveStateBad", 42);'
```
delete variable:
```   
homegear -e 'runcommand $hg->deleteSystemVariable("maxValveStateBad");'
```

I don't know which other information might be interesting. Please ask.

Same problem here:

`2016-12-14 18:42:26.545 [ERROR] [very.HomematicDeviceDiscoveryService] -
java.lang.NullPointerException
        at org.openhab.binding.homematic.internal.communicator.AbstractHomematicGateway.createGatewayDevice(AbstractHomematicGateway.java:655)[181:org.openhab.binding.homematic:2.0.0.201612120809]
        at org.openhab.binding.homematic.internal.communicator.AbstractHomematicGateway.getDeviceDescriptions(AbstractHomematicGateway.java:420)[181:org.openhab.binding.homematic:2.0.0.201612120809]
        at org.openhab.binding.homematic.internal.communicator.AbstractHomematicGateway.loadAllDeviceMetadata(AbstractHomematicGateway.java:358)[181:org.openhab.binding.homematic:2.0.0.201612120809]
        at org.openhab.binding.homematic.discovery.HomematicDeviceDiscoveryService$1.run(HomematicDeviceDiscoveryService.java:108)[181:org.openhab.binding.homematic:2.0.0.201612120809]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_65]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_65]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)[:1.8.0_65]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)[:1.8.0_65]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_65]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_65]
        at java.lang.Thread.run(Thread.java:745)[:1.8.0_65]`

But I think it have nothing to do with the system variables. I have deleted all my system variables, but the problem is not solved.

Same problem with homegear 6.7-1545 and openHab2-online 2.0.0~20161222170724. Looks like the homematic binding  can't read the homegear system variables.

```
2016-12-22 20:46:07.515 [ERROR] [very.HomematicDeviceDiscoveryService] -
java.lang.NullPointerException
        at org.openhab.binding.homematic.type.MetadataUtils.createBigDecimal(MetadataUtils.java:254)[180:org.openhab.binding.homematic:2.0.0.201612221443]
        at org.openhab.binding.homematic.type.HomematicTypeGeneratorImpl.createChannelType(HomematicTypeGeneratorImpl.java:246)[180:org.openhab.binding.homematic:2.0.0.201612221443]
        at org.openhab.binding.homematic.type.HomematicTypeGeneratorImpl.generate(HomematicTypeGeneratorImpl.java:129)[180:org.openhab.binding.homematic:2.0.0.201612221443]
        at org.openhab.binding.homematic.handler.HomematicBridgeHandler.onDeviceLoaded(HomematicBridgeHandler.java:285)[180:org.openhab.binding.homematic:2.0.0.201612221443]
        at org.openhab.binding.homematic.internal.communicator.AbstractHomematicGateway.loadAllDeviceMetadata(AbstractHomematicGateway.java:400)[180:org.openhab.binding.homematic:2.0.0.2016122
21443]
        at org.openhab.binding.homematic.discovery.HomematicDeviceDiscoveryService$1.run(HomematicDeviceDiscoveryService.java:108)[180:org.openhab.binding.homematic:2.0.0.201612221443]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_65]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_65]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)[:1.8.0_65]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)[:1.8.0_65]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_65]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_65]
        at java.lang.Thread.run(Thread.java:745)[:1.8.0_65]
```

--- a/addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/type/HomematicTypeGeneratorImpl.java
+++ b/addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/type/HomematicTypeGeneratorImpl.java
@@ -254,10 +254,8 @@ public class HomematicTypeGeneratorImpl implements HomematicTypeGenerator {
                 BigDecimal min = MetadataUtils.createBigDecimal(dp.getMinValue());
                 BigDecimal max = MetadataUtils.createBigDecimal(dp.getMaxValue());
 
-                BigDecimal step = null;
-                if (dp.getStep() != null) {
-                    step = MetadataUtils.createBigDecimal(dp.getStep());
-                } else {
+                BigDecimal step = MetadataUtils.createBigDecimal(dp.getStep());
+                if (step == null) {
                     step = MetadataUtils.createBigDecimal(dp.isFloatType() ? new Float(0.1) : 1L);
                 }
                 state = new StateDescription(min, max, step, MetadataUtils.getStatePattern(dp), dp.isReadOnly(),

--- a/addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/type/MetadataUtils.java
+++ b/addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/type/MetadataUtils.java
@@ -251,6 +251,9 @@ public class MetadataUtils {
      * Helper method for creating a BigDecimal.
      */
     public static BigDecimal createBigDecimal(Number number) {
+        if (number == null) {
+            return null;
+        }
         try {
             return new BigDecimal(number.toString());
         } catch (Exception ex) {

